### DISCUSSION TITLE
Suggestion: no more return self

### DISCUSSION THREAD

**Discussion by MaxHalford** [online-ml/river#935] [Ideas]
I think it would make sense to remove all the `return self` from the update methods of every class we have. I added this initially because that's what scikit-learn does, and scikit-learn had a big influence on how I did things. But the more I think, the more I see this as noise. Some arguments:

- There are ways of running a training loop. Either `model = model.learn_one(x, y)`, either `model.learn_one(x, y)`. Both do the same thing. I rather have one way of doing things.
- We are kinda forced to do `model = model.learn_one(x, y)` in docstrings, else the model is expected to be part of the docstring return.
- Having a `return self` allows chaining method calls, such `model.learn_one(x, y).learn_one(x, y).predict_one(x)`. But I've never seen that usage pattern.
- `model = model.learn_one(x, y)` gives a false impression that `model.learn_one(x, y)` is a pure function call with no side-effects. But it isn't.

Of course, I'd be happy to implement this change if we agree on it. I suggest we take a vote between @smastelini, @jacobmontiel, and I. @jacobmontiel is busy at the moment, so let's agree on the change if @smastelini votes in favor.

**Comment by smastelini** [comment]
I agree with that. It always looked weird to me, since the skmultiflow times. 

Chaining method calls although useful adds unnecessary clutter. It's not that much used, anyhow. 

You have my vote, @MaxHalford :)
